SDL/MacOSBarItems.mm

```diff
@@ -0,0 +1,326 @@
+    
+    NSMenuItem *gpuBackendItem = [[NSMenuItem alloc] initWithTitle:GRAPHICS_LOCALIZED("Backend") action:nil keyEquivalent:@""];
+    
+    std::vector<GPUBackend> allowed = [self allowedGPUBackends];
+    for (int i = 0; i < allowed.size(); i++) {
+        NSMenuItem *backendMenuItem = [[NSMenuItem alloc] initWithTitle:@(GPUBackendToString(allowed[i]).c_str()) action: @selector(setCurrentGPUBackend:) keyEquivalent: @""];
+        backendMenuItem.tag = i;
+        backendMenuItem.target = self;
+        backendMenuItem.state = [self controlStateForBool: g_Config.iGPUBackend == (int)allowed[i]];
+        [backendsMenu addItem:backendMenuItem];
+    }
+    
+    gpuBackendItem.submenu = backendsMenu;
+    [parent addItem:gpuBackendItem];
+    
+    [parent addItem:[NSMenuItem separatorItem]];
+    
+#define MENU_ITEM(variableName, localizedTitleName, SEL, ConfigurationValueName) \
+NSMenuItem *variableName = [[NSMenuItem alloc] initWithTitle:GRAPHICS_LOCALIZED(localizedTitleName) action:SEL keyEquivalent:@""]; \
+variableName.target = self; \
+variableName.state = [self controlStateForBool: ConfigurationValueName];
+    
+    MENU_ITEM(softwareRendering, "Software Rendering", @selector(toggleSoftwareRendering:), g_Config.bSoftwareRendering)
+    [parent addItem:softwareRendering];
+    
+    MENU_ITEM(vsyncItem, "VSync", @selector(toggleVSync:), g_Config.bVSync)
+    [parent addItem:vsyncItem];
+    
+    MENU_ITEM(fullScreenItem, "Fullscreen", @selector(toggleFullScreen:), g_Config.bFullScreen)
+    [parent addItem:fullScreenItem];
+    
+    [parent addItem:[NSMenuItem separatorItem]];
+    
+    MENU_ITEM(autoFrameSkip, "Auto FrameSkip", @selector(toggleAutoFrameSkip:), g_Config.bAutoFrameSkip)
+    [parent addItem:autoFrameSkip];
+    
+    [parent addItem:[NSMenuItem separatorItem]];
+    
+    MENU_ITEM(fpsCounterItem, "Show FPS Counter", @selector(setToggleShowCounterItem:), g_Config.iShowStatusFlags & (int)ShowStatusFlags::FPS_COUNTER)
+    fpsCounterItem.tag = (int)ShowStatusFlags::FPS_COUNTER;
+    
+    MENU_ITEM(speedCounterItem, "Show Speed", @selector(setToggleShowCounterItem:), g_Config.iShowStatusFlags & (int)ShowStatusFlags::SPEED_COUNTER)
+    speedCounterItem.tag = (int)ShowStatusFlags::SPEED_COUNTER;
+    
+    MENU_ITEM(batteryPercentItem, "Show battery %", @selector(setToggleShowCounterItem:), g_Config.iShowStatusFlags & (int)ShowStatusFlags::BATTERY_PERCENT)
+    batteryPercentItem.tag = (int)ShowStatusFlags::BATTERY_PERCENT;
+    
+    [parent addItem:[NSMenuItem separatorItem]];
+    [parent addItem:fpsCounterItem];
+    [parent addItem:speedCounterItem];
+    [parent addItem:batteryPercentItem];
+    
+    [[NSNotificationCenter defaultCenter] addObserverForName:@"ConfigDidChange" object:nil queue:nil usingBlock:^(NSNotification * _Nonnull note) {
+        NSString *value = [note object];
+        // NOTE: Though it may seem like it,
+        // the next few lines were not written by yandere dev
```

Comment:
?

Reply:
A joke reference to Yandere simulator developer who was a meme about DRY violation and exhaustive if/else coding. 

Shouldn't be a commend imho as I don't really like the shaming :)

Pic for reference: https://i.redd.it/gyubzyq87xmx.png